Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/io/writer.go

```diff
@@ -56,10 +58,11 @@ func (writer *NsenterWriter) WriteFile(filename string, data []byte, perm os.Fil
 		"--",
 	}
 
-	echo_args := append(base_args, "sh", "-c",
-		fmt.Sprintf("echo %q | cat > %s", data, filename))
+	echo_args := append(base_args, "sh", "-c", fmt.Sprintf("cat > %s", filename))
```

Comment:
There is only one code path today that invokes this method, the inputs are the volume mount path (not user input) and secret data key (user input) but secret data key is strongly protected.  Opened https://github.com/kubernetes/kubernetes/issues/16971 to track resolution of that.  For this code, in this spot, we're safe.